### PR TITLE
Ar/folderstruct

### PR DESCRIPTION
**Pull request by AlexisRenchon**
## Purpose 

This is a small PR that we can hopefully merge soon (this week?) if you think it is a good idea (it is okay if you think it is not!)

The idea is to have a documentation page that describes the structure of `ClimaLSM` repository. When I go through a repo on GitHub, I always try to understand what each folder and each file is meant to do, it helps me navigate and understand how the repo is structured and work... I imagine it could be very helpful for newcomers, whether they are students or advanced scientist. It can also help us organize the repo better. 

This PR goes hand-to-hand with PR #220 but is small and easy to merge (it is just a doc page).  

## To-do

Just go through https://github.com/CliMA/ClimaLSM.jl/blob/ar/folderstruct/docs/src/folderstruct.md, it is easy to read it there, and edit it as you see fit. 

## Content

Currently it is very simplified. We can decide how much details we go into, let's discuss on this PR below! 

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

File: docs/src/folderstruct.md
Comment (kmdeck) [inline docs/src/folderstruct.md:5]: can remove the <br> (it appears for me in the rendered page?)

**Comment by kmdeck** [comment]
hey Alexis! I think this is a nice idea, I'll leave a few comments in review.

We might just be careful to not describe too much in detail because then this doc page would need to get updated often, and it is easy to forget to. But right now it looks like a good level!

File: docs/src/folderstruct.md
Comment (kmdeck) [inline docs/src/folderstruct.md:10]: "work as intended"

File: docs/src/folderstruct.md
Comment (kmdeck) [inline docs/src/folderstruct.md:13]: buildkite (small typo)

We can also note that these buildkite runs are carried out as part of CI and must run without error in order to merge a PR.

File: docs/src/folderstruct.md
Comment (kmdeck) [inline docs/src/folderstruct.md:14]: i think right now this just has the julia formatter? Not totally sure how to describe the purpose of this folder.

File: docs/src/folderstruct.md
Comment (juliasloan25) [reply docs/src/folderstruct.md:5]: I think you mean to remove the \<br\> (it got removed from the markdown comment). It doesn't look like it shows up in the rendered page for me though (looking at "Preview")

File: docs/src/folderstruct.md
Comment (kmdeck) [inline docs/src/folderstruct.md:39]: ozark_domain.jl

File: docs/src/folderstruct.md
Comment (kmdeck) [inline docs/src/folderstruct.md:41]: here we may not want to give the specific value, because then any PR which changes that value also has to remember to update this docs page

File: docs/src/folderstruct.md
Comment (kmdeck) [inline docs/src/folderstruct.md:48]: for PDE domains

Comment (kmdeck) [inline docs/src/folderstruct.md:48]: "which will be used to create a function which updates the auxiliary parameters, stored in the vector `p`"

File: docs/src/folderstruct.md
Comment (kmdeck) [inline docs/src/folderstruct.md:54]: This is how we interact with CLIMAParameters, which describes a single interface and parameter set that all Clima models use. This ensures all components of the Earth System Model use the same values for all parameters, and that the same parameter calibration tools can be used to calibrate all models, either jointly or independently.

File: docs/src/folderstruct.md
Comment (juliasloan25) [inline docs/src/folderstruct.md:55]: Contains code used to regrid fields between different meshes, using ClimaCore data structures. This code uses ClimaCoreTempestRemap.jl under the hood and is largely copied from ClimaCoupler.jl. It may be moved to a different CliMA repository in the near future.

This code could also probably be moved to SharedUtilities

File: docs/src/folderstruct.md
Comment (juliasloan25) [inline docs/src/folderstruct.md:52]: There are also 3 other files - `pond_soil_model.jl`, `soil_energy_hydrology_biogeochemistry.jl`, and `soil_plant_hydrology_model.jl`. Are they not included here because we're planning to move them to Standalone/Integrated folders?

**Review by juliasloan25 — COMMENTED** [review]
This is a great idea! Thanks for doing this, Alexis. I left a few comments/suggestions. Another thought is maybe we could rename the file itself to `folderstructure.md` - I think that would be a bit more clear

File: docs/src/folderstruct.md
Comment (AlexisRenchon) [reply docs/src/folderstruct.md:54]: I will leave this comment unresolved, as maybe we should indeed move Parameters.jl into SharedUtilities. (We could do it in this PR?) 
What do you think @kmdeck?

File: docs/src/folderstruct.md
Comment (AlexisRenchon) [reply docs/src/folderstruct.md:55]: Same comment here as for Parameters.jl. Move to SharedUtilities @kmdeck?

File: docs/src/folderstruct.md
Comment (AlexisRenchon) [reply docs/src/folderstruct.md:52]: Yes they will be in integrated folder. 
I wonder if it is okay to merge this PR first though, even if it will have some inconsistencies until we actually merge #220 (that needs some work before being merged)

**Comment by AlexisRenchon** [comment]
Hello @juliasloan25, @kmdeck, I have been through your comments and edited the PR accordingly! 

I see on the docs preview that there are still some small issues to fix (e.g., some text in italic not intentionally).

Julia mentioned moving `/src/Parameters.jl` and `/src/Regridder.jl` to `src/SharedUtilities/`, which would be neat from an organization perspective (less files in /src - /src would just be SharedUtilities, Standalone, Integrated folder and ClimaLSM.jl!).   

Moving files is easy, but that means we need to adjust the path to these files in the scripts calling them. I can do it in the next commit if you agree it is a good idea @kmdeck. 

After that, maybe someone else could do a final review before we merge? 

Then I will also finalize #220 - hopefully all done early next week!

File: docs/src/folderstruct.md
Comment (kmdeck) [reply docs/src/folderstruct.md]: I think it is a good idea!

Comment (kmdeck) [reply docs/src/folderstruct.md]: we can do it in this PR, I think!

File: docs/src/folderstruct.md
Comment (kmdeck) [reply docs/src/folderstruct.md]: I think it is ok! the intention is to follow up with the restructure relatively quickly, so all good.

File: docs/src/folderstruct.md
Comment (AlexisRenchon) [reply docs/src/folderstruct.md]: ok! will do in next commit =)

File: docs/src/folderstructure.md
Comment (AlexisRenchon) [reply docs/src/folderstructure.md:50]: I think it is a good thing to mention!

**Comment by AlexisRenchon** [comment]
I think we are ready to merge!

I moved `src/Regridder.jl` and `src/Parameters.jl` to `src/SharedUtilities/Regridder.jl` and `src/SharedUtilities/Parameters.jl`. I corrected the path to these files in `src/ClimaLSM.jl`. I believe it should be all set. If it passes CI it should mean it is okay?

**Review by kmdeck — APPROVED** [review]
looks good to me! But first, can we squash these 17 commits into 1? I can help with that if you want to pair-program over zoom!

File: docs/src/folderstructure.md
Comment (kmdeck) [inline docs/src/folderstructure.md:53]: oh! one other quick fix - these descriptions can now be moved under "SharedUtilities", since you moved them :)

**Pull request merged**
